src/Engines/Processors/iShaderProcessor.ts

```diff
@@ -1,14 +1,24 @@
+import { ShaderLanguage } from "../../Materials/shaderLanguage";
 import { Nullable } from "../../types";
 import { ShaderProcessingContext } from "./shaderProcessingOptions";
 
 declare type ThinEngine = import("../thinEngine").ThinEngine;
 
 /** @hidden */
 export interface IShaderProcessor {
+    shaderLanguage: ShaderLanguage;
+
+    uniformRegexp?: RegExp;

```

Comment:
last one, I wondering why we need dynamic regex at the interface level here

Reply:
I am actually not seeing where they are set but the PR is huge so I have an excuse ;-)

Reply:
Those regexp are used by both `ShaderCodeNode` and the webgpuShaderProcessorGLSL/webgpuShaderProcessorWGSL classes so I factored them in the interface. If `undefined` for one type of object, it means `ShaderCodeNode` should not called the corresponding interface method for this type of object (`uniformProcessor`, `textureProcessor`, etc).